robot_controllers/src/follow_joint_trajectory.cpp

```diff
@@ -419,8 +419,10 @@ void FollowJointTrajectoryController::executeCb(const control_msgs::FollowJointT
       // use the generated trajectory
       executable_trajectory = new_trajectory;
 
-      // if this hasn't started yet, need to insert current position
-      if (goal->trajectory.points[0].time_from_start.toSec() > 0.0)
+      // if this hasn't started yet or if the header stamp is in the future,
+      // need to insert current position
+      if (goal->trajectory.points[0].time_from_start.toSec() > 0.0 ||
+          goal->trajectory.header.stamp.toSec() > ros::Time::now().toSec())
```

Comment:
So while this logic might work -- I think it might be clearer to use "executable_trajectory.points[0].time > ros::Time::now().toSec()" in the "if" statement.

If you look at trajectoryFromMsg() -- the start time is already baked into the executable_trajectory (if the header has time 0, it uses ros::Time::now() -- which is I believe what happens with most MoveIt trajectories, and so the only case where this was needed with pure MoveIt commands was if the first point was not at 0.0).

Reply:
Thank you for your feedback!
I will switch to use `executable_trajectory.points[0].time` and test it in my environment.